docs/guides/writing-advanced-validations.mdx

```diff
@@ -4,6 +4,489 @@ sidebar_label: 'Writing Advanced Validations'
+    <div className="col col--6">
+        <img src={UseAOESkillSupplement1} alt="Target heavy pot"/>
+    </div>
+    <div className="col col--6">
+        <img src={UseAOESkillSupplement2} alt="First skill changes to 'Pick Up'"/>
+    </div>
+</div>
+
+<img src={UseAOESkillStep} alt="At least one Imp exists" className="margin-bottom--md" style={{width: "30rem", display: "block", margin: "auto"}}/>
+
+Enemies play stagger animations when they're hit in Boss Room.
+We can validate that all three Imps were hit by the AOE skill by asserting that three `Hitreact Minor` objects appear in the scene.
+
+import AOEImpStaggerStep from "./img/writing-advanced-validations/using-character-abilities/aoe-imp-stagger-step.png"
+
+<figure  className="margin--none margin-bottom--md text--center">
+    <img src={AOEImpStaggerStep} alt="Three Imps were hit and staggered" style={{width: "30rem"}}/>
+    <figcaption>
+        <i>Select the Imp <code>Hitreact</code> animation specifically! The player has one too.</i>
+    </figcaption>
+</figure>
+
+:::info Building Appropriate Validations
+
+Why are we indirectly validating that Imps were hit using stagger effects rather than raw health values?
+
+In this Gameplay Session, we cast the AOE skill three times.
+This means we probably want to validate that _every_ skill cast causes some amount of damage to each Imp.
+We could try to capture the initial health value of an Imp using the "Store" step,
+but this will only work before any Imps have taken damage.
+The "Store" step only succeeds when it can definitively determine what value to capture from a field.
+If there is only one Imp in the scene, or if every Imp has the same amount of health, then the
+"Store" step knows what value it should store.
+However, if we have multiple Imps with various amounts of health remaining, then the "Store" step
+won't be able to determine which of those values should be stored.
+This will cause the step to fail.
+
+Alternatively, we could use the "Wait for State" step to assert that three Imps drop below a specific health value
+(100, 75, 25, etc.).
+This may work for your game if you know the max health value for an enemy and the minimum amount of damage a skill should deal to that enemy.
+However, Boss Room doesn't expose this data directly to our SDK.
+Health bars are rectangles whose fill levels are determined by remaining percentage of health,
+but the fill level doesn't translate well to a meaningful health value.
+
+There may be many different ways to achieve a Scenario for your game.
+The "best" configuration of steps will depend heavily on your game state.
+
+:::
+
+Depending on the goals of your Scenario, you may want to repeat these steps for each cast of the AOE skill.
+You can repeat the last sequence of steps two more times to achieve this effect.
+Or, you may simply want to validate that the skill is mechanically sound and eventually results in the decimation of your enemies.
```

Comment:
Rewrote

```
For the sake of this example, we'll only write steps for the first AOE effect.
However, it would be good practice for this Scenario to repeat the same steps for each instance of the skill
to validate that it produces consistent results.
You can achieve this by repeating the previous sequence of steps two more times.

Finally, wrap up the Scenario by validating that all three Imps die at the same time and produce an `FX_ImpDefeat` animation.
```